docs/faq/add_sparse_op_in_backend.md

```diff
@@ -0,0 +1,439 @@
+storage type respectively. 
+- Lines 13-18: If both the input storage type and the output storage type
+are "csr" and c is 0.0, invoke the "csr" implementation. Otherwise,
+an exception will be thrown with detailed information about the unimplemented
+operator arguments.
+- Lines 20-25: Function definition for the "csr" implementation of the `quadratic`
+operator.
+- Lines 26-28: Declare a few namespaces used in the current function scope.
+Note that the `csr::kIdx` is for the access to the `indices` array of
+all auxiliary arrays, while `csr::kIndPtr` is for the access to the `indptr`
+array. 
+- Line 29-30: Check the provided `req` of the operator. If `req` is `kNullOp`,
+no work is required. Since the output of this operator is a "csr" NDArray,
+whose memory has to be allocated at runtime, only `kWriteTo` is allowed.
+Both `kAddTo` and `kWriteInplace` usually are not supported when the
+output is sparse.
+- Line 31: Get the `stream` of the context for serializing asynchronous executions.
+- Lines 32-35: Before we access the `data`, `indices` and `indptr` arrays
+to compute the result, we first check if these arrays are empty. If so,
+we set the output to be zeros.
+The `storage_initialized()` method returns true if a sparse NDArray
+contains at least one element in its data and indices array; it returns false
+otherwise.
+- Line 36: Get the number of elements stored in the input and store
+it in variable `nnz`. The `storage_shape()` method returns the shape of the
+`data` array of a sparse NDArray.
+- Line 37: Get the number of rows of the output and store it in variable `num_rows`.
+- Line 38: Allocate memory for the data array and auxiliary arrays. For a CSRNDArray
+of shape (M, N) storing K elements, it has a `data` array of length K, an `indices` array 
+of length K and an `indptr` array of length (M + 1).
+The `CheckAndAlloc` method takes the shape of auxiliary arrays as the input, and allocates
+the memory for the data array and auxiliary arrays. It is not necessary to provie
+the shape of the data array, as it can be inferred from shapes of auxilary arrays.
+- Line 39-54: This is the place where the values of output data array and auxiliary arrays
+are computed. 
+The macros `MSHADOW_TYPE_SWITCH` and `MXNET_ASSIGN_REQ_SWITCH` enable
+the code block to work for all the supported data types and `req` types in MXNet.
+For this operator, since the transformation only happens on the data array,
+we simply invoke the quadratic operator kernel [quadratic_forward](https://mxnet.incubator.apache.org/faq/add_op_in_backend.html#forward-function) via `Kernel::Launch`.
+For the `indices` and `indptr` arrays, we just copy the values from the inputs.
+This way, a complete output CSRNDArray is computed. 
+
+### Operator Registration
+Finally let's extend the operator registration logic to expose `sparse.quadratic`
+to frontend. Below is the extended registration code in `quadratic_op.cc`:
+```cpp
```

Comment:
Having line numbers here will you to reference to them later, and help readers to understand the content.

Reply:
Sure I'll add it